### DISCUSSION TITLE
Centralized Messaging

### DISCUSSION THREAD

**Discussion by kasugaijin** [rubyforgood/pet-rescue#468] [Ideas]
**Background:** 
Organization staff usually receive messages from a number of different channels, e.g., text, email, WhatsApp, Instagram, Facebook etc., and replying to, and keeping track of these messages is a big time sink. 

**Centralized Messaging:**
- A feature where staff can receive, and respond to, messages from various channels e.g., (email, text, popular platforms)
- Messages can come from both organization registered (e.g., adopters and fosters) and non-registered individuals
- Phase I - Messaging feature that allows staff to receive messages from, and send messages to email, text, (other apps?)
- Phase II - allow staff to send notifications to adopters/fosters e.g., for vaccine reminders
- Ideally, integrations with each messaging channel are robust and not likely to break with third-party changes to APIs etc.

I am not sure where we should draw the line in terms of scope for what channels we want to integrate with. I think at a minimum, email and maybe text are a core requirement. But, if we can integrate with popular platforms, also, even better. Let's discuss.

**Comment by maebeale** [comment]
This feature will dramatically reduce org staff time!!!!! All orgs interviewed talked about how much of their time is spent being switchboard operator between themselves, their peer staff, volunteers, fosters, adopters, and prospective people who may want to be involved as any of those categories and or a provider/facilitator of sending a new pet their way. Orgs advertise pets through many channels bc this whole system is dependent on foster/adopter recruitment. The more humans they can engage, the more pets will new saved. This means messages are coming in from website, email, text, fb, insta (platforms w pics entice more humans, so orgs are moving to wherever they're getting the most traction). Then, the staff are communicating w the humans, connecting them with each other so all ensure proper health care and good transitions and experiences are had by all. Maybe all that is already implied, but, just wanted to share my thoughts on what this feature will actually do for an org's ability to serve pets better by engaging w humans in ways that are streamlined for staff. Most staff at these orgs are running so fast and trying to track so many details through so many channels that their heads are spinning and details are getting written on scraps of paper bc they don't have a way to convert the info to data. 

There's an additional consideration here -- the conversations are not simply between staff and 1 external person. 

The reason they're using so many channels and creating whatsapp groups is bc in the Foster scenario they're wanting to foster (haha) relationships between the Fosterer and the future-but-already-assigned Adopter and their pet. So, it's a 3-way conversation (and I think irl there's 1+ parties involved on the Foster and the Adopter sides). So they are talking to each other and the staff is monitoring and chiming in a needed. 

There are 1:1 phases of the conversations at beginning of both Foster and Adopter flows, then the all-hands-until-adoption and then the staff told us that post-adoption the Adopter is keeping the Foster parent in the loop on the pet before eventually that falls off and things go back to Fosterer and Adopter are communicating 1:1 w staff as needed. 

So, there's need for basic message relay for the million random 1:1 conversations, but there's a clear case for a second kind of comms.

I think phase 1 is there's one chat for every registered pet. The staff can invite relevant people as things progress. 
- pet registration it's staff talking with other staff and volunteers about onboarding a pet
- fosterer is added (if applicable, bc not all orgs will do fostering. In that case the org itself is the foster, ensuring vaccinations etc)
- Adopter is added
- adoption transfer of pet from foster to adopter gets scheduled and then happens (this is a lot of three convo and coordination staff are doing between people that don't both have fb or text or whatsapp)
- Adopter keeps foster and org in the loop
- Adopter keeps getting reminders from org but prob fosterer isn't needed in convo anymore at which time staff removes foster and keeps pet convo between Adopter and staff


Some dreamy future might include an option to click a button on a message in a chat to convert it to log, for ex, that a vaccination happened, that a meeting was scheduled, that a transfer took place, etc. Or NLP suggests it...

@kasugaijin does this match your understanding as well?

**Reply by kasugaijin** [reply]
You remembered a lot more than I did from the discussiona @maebeale thank you for the additional details :)
I agree with the above. I want to check in with the stakeholders about the **need** for more than a 1 to 1 discussion. I want to know how much of this is a nice to have versus need to have and will report back here.
I think the main thing is we build this to MVP level, i.e., works and is useful but not fully fledged, but design in a way that can be modified.

**Reply by kasugaijin** [reply]
Update - the stakeholders say that 1 to 1 is fine for messaging. They like to keep a tight leash (pun intended) on dialogue because people and ideas can start to get out of control and harder to manage. @maebeale 

They also sometime have a group chat for just staff or just fosterers. Though I am not yet convinced this is best handled in app or another existing platform like slack. Perhaps it’s a reasonable later phase to explore.

**Comment by ghost** [comment]
Is the vision to have a full-fledged messaging interface built out in the app?  Or would org employees use an existing platform (like WhatsApp, or FB Messenger, or something else) for sending/receiving?

IMO it would be easier to have the app integrate with an existing platform than it would be to build out messaging within the app itself.

For example, adopter fills out an application to adopt a specific pet, the app spins up a new channel in Slack (or starts a new conversation in WhatsApp, something like that) and invites the right people.

I think what I'm getting at is: does the app need to be the thing through which all the messages flow?  Or can it just be a tool that helps facilitate conversations in a more mature platform?

**Reply by kasugaijin** [reply]
I agree that we should first explore whether the 'simpler' solution will provide the value we are looking for, i.e., integrate with exsiting platforms instead of build something from scratch. Especially for MVP. 

I think in order to answer this question, we really need to involve the stakeholders to validate our assumptions.

The assumptions I have are:
- Pet Rescue orgs receive/send messages via email, text, social media apps
- Pet Rescue orgs have a hard time keeping track of conversations with people across each different medium, e.g., where was that dicussion for Fluffy? Was it email or WhatsApp, or Instagram? Or Facebook?
- People are most comfortable using the apps they prefer to use e.g., email, Instagram, WhatsApp etc.
- Pet Rescue Orgs are flexible and allow people to use their preferred method of communication (within reason)
- We do not need to build our own messaging application to replace people's preferred apps
- The value we add is a central point on the app where staff can easily access conversations to receive/send messages
- To the last point, this is where the implementation needs to be defined - do we build so staff send/receive messages IN app? Or do we, like @uoodsq said, point to the conversations on other apps? I think the latter could be sufficient. I think that would take a lot of the overhead out of tracking conversations if staff can go to the "Message Center" in app, and filter either on Pet name or Person name to find a link to the conversation in another platform. It becomes more of a 'conversation tracker', than an actual messaging app.

If we can refine/develop this assumption list, I can then meet with stakeholders to validate. Does that sound like a reasonable approach?
